sdk/tables/azure-data-tables/src/test/java/com/azure/data/tables/TablesAsyncClientTest.java

```diff
@@ -301,60 +303,90 @@ void updateEntityWithResponseAsync(UpdateMode mode) {
+        tableClient.createEntity(new TableEntity(partitionKeyValue, rowKeyValue)).block(TIMEOUT);
+        tableClient.createEntity(new TableEntity(partitionKeyValue, rowKeyValue2)).block(TIMEOUT);
 
         // Act & Assert
         StepVerifier.create(tableClient.listEntities(options))
             .assertNext(returnEntity -> {
                 assertEquals(partitionKeyValue, returnEntity.getPartitionKey());
-                assertEquals(entity.getRowKey(), returnEntity.getRowKey());
+                assertEquals(rowKeyValue, returnEntity.getRowKey());
             })
+            .expectNextCount(0)
+            .thenConsumeWhile(x -> true)
             .expectComplete()
             .verify();
     }
 
-    @Disabled("List not working yet.")
     @Test
-    void listEntityWithSelectAsync() {
+    @Tag("ListEntities")
+    void listEntitiesWithSelectAsync() {
         // Arrange
         final String partitionKeyValue = testResourceNamer.randomName("partitionKey", 20);
         final String rowKeyValue = testResourceNamer.randomName("rowKey", 20);
         final TableEntity entity = new TableEntity(partitionKeyValue, rowKeyValue)
             .addProperty("propertyC", "valueC")
             .addProperty("propertyD", "valueD");
         ListEntitiesOptions options = new ListEntitiesOptions()
-            .setFilter("PartitionKey eq '" + entity.getPartitionKey() + "'")
             .setSelect("propertyC");
         tableClient.createEntity(entity).block(TIMEOUT);
 
         // Act & Assert
         StepVerifier.create(tableClient.listEntities(options))
             .assertNext(returnEntity -> {
-                assertEquals(entity.getRowKey(), returnEntity.getRowKey());
-                assertEquals(entity.getPartitionKey(), returnEntity.getPartitionKey());
+                assertNull(returnEntity.getRowKey());
+                assertNull(returnEntity.getPartitionKey());
                 assertEquals("valueC", returnEntity.getProperties().get("propertyC"));
-                assertEquals(3, returnEntity.getProperties().size());
+                assertNull(returnEntity.getProperties().get("propertyD"));
             })
             .expectComplete()
             .verify();
     }
 
     @Test
-    void listEntityWithTopAsync() {
+    @Tag("ListEntities")
+    void listEntitiesWithTopAsync() {
         // Arrange
-        ListEntitiesOptions options = new ListEntitiesOptions()
-            .setTop(1);
+        final String partitionKeyValue = testResourceNamer.randomName("partitionKey", 20);
+        final String rowKeyValue = testResourceNamer.randomName("rowKey", 20);
+        final String rowKeyValue2 = testResourceNamer.randomName("rowKey", 20);
+        final String rowKeyValue3 = testResourceNamer.randomName("rowKey", 20);
+        ListEntitiesOptions options = new ListEntitiesOptions().setTop(2);
+        tableClient.createEntity(new TableEntity(partitionKeyValue, rowKeyValue)).block(TIMEOUT);
```

Comment:
You can use the following, this is like Task.whenAll that completes when all three are done.
```java
Mono.when(
    tableClient.createEntity(new TableEntity(partitionKeyValue, rowKeyValue)),
    tableClient.createEntity(new TableEntity(partitionKeyValue, rowKeyValue2)),
    tableClient.createEntity(new TableEntity(partitionKeyValue, rowKeyValue3))).block(TIMEOUT);
```

Reply:
Same with those other instances.

Reply:
Fixed